Eugene/MovieSearch/my-app/src/components/MainPage/search-panel/SearchPanel.tsx

```diff
@@ -0,0 +1,213 @@
+}
+
+interface ISearchPanelState {
+  input: string;
+  searchBy: SearchBy;
+}
+
+enum SearchBy {
+  Title = "title",
+  Genre = "genres",
+}
+
+function SearchPanel(props: ISearchPanelProps) {
+  const [input, setInput] = useState("");
+  const [searchBy, setSearchBy] = useState(SearchBy.Title);
+
+  useEffect(() => {
+    checkParams();
+  }, [props.history.location]);
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  };
+
+  const handleSearchByBtn = (btnType: SearchBy) => {
+    setInput("");
+    setSearchBy(btnType);
+  };
+
+  const checkParams = () => {
+    const searchParams = props.location.search.slice(1);
+    const parsed = parse(searchParams) as {
+      search: string;
+      searchBy: string;
+    };
+
+    setInput(parsed.search || "");
+    if (parsed.searchBy === SearchBy.Genre) {
+      setSearchBy(SearchBy.Genre);
+    } else {
+      setSearchBy(SearchBy.Title);
+    }
+  };
+
+  const handleSearchSubmit = () => {
+    props.handleSearch(input, searchBy);
+  };
+
+  const keyPress = (e: React.KeyboardEvent) => {
+    if (e.key === "Enter") {
+      props.handleSearch(input, searchBy);
+    }
+  };
+
+  return (
+    <div className="search-panel-wrapper">
+      <h2 className="search-panel-label">FIND YOUR MOVIE</h2>
+      <input
+        onKeyPress={keyPress}
+        onChange={handleSearchChange}
+        type="text"
+        name=""
+        value={input}
+        className="search-panel-input"
+      />
+      <div className="search-by">
+        <div className="search-by-btns">
+          <p>SEARCH BY</p>
+          <button
+            onClick={() => handleSearchByBtn(SearchBy.Title)}
+            className={
+              searchBy === SearchBy.Title
+                ? "search-by-btn-active"
+                : "search-by-btn"
+            }
+          >
+            TITLE
+          </button>
+          <button
+            onClick={() => handleSearchByBtn(SearchBy.Genre)}
+            className={
+              searchBy === SearchBy.Genre
+                ? "search-by-btn-active"
+                : "search-by-btn"
+            }
+          >
+            GENRE
+          </button>
+        </div>
+        <div>
+          <button onClick={handleSearchSubmit} className="search-btn">
+            SEARCH
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+// class SearchPanel extends React.Component<
+//   ISearchPanelProps,
+//   ISearchPanelState
+// > {
+//   state: ISearchPanelState = {
+//     input: "",
+//     searchBy: SearchBy.Title,
+//   };
+
+//   componentDidMount() {
+//     this.checkParams();
+//   }
+
+//   componentDidUpdate(prevprops: RouteComponentProps) {
+//     if (this.props.location.search !== prevprops.location.search) {
+//       this.checkParams();
+//     }
+//   }
+
+//   handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+//     this.setState({
```

Comment:
I understand that you've left it for study purpose, but never leave such things on a real project! 

I would suggest you to make a separate branch from hooks version. 
1. Just do the checkout from current "main" branch and remove all of the redundant code. (git checkout -b "hooks"). 
2. Go back to your "main" branch and reset last commit with hooks. 
(git reset --hard HEAD^1).
So you will have two separate branches with two version of react and you won't need to comment.